src/runtime/scheduler.go

```diff
@@ -134,20 +176,43 @@ func scheduler() {
 			runqueue.Push(t)
 		}
 
+		// Check for expired timers to trigger.
+		if timerQueue != nil && now >= timerQueue.whenTicks() {
+			scheduleLog("--- timer awoke")
+			// Pop timer from queue.
+			tn := timerQueue
+			timerQueue = tn.next
+			tn.next = nil
+			// Run the callback stored in this timer node.
+			tn.callback(tn)
```

Comment:
I don't think this is really safe with coroutine lowering right now?

Any user code can define a function value of the type `func(interface{}, uintptr)` which I think is sufficient to break this?

Reply:
I'm afraid you're right. I'll have to think a bit about how to fix this.

Thank you for taking a look. I hadn't thought of this.

Reply:
So on coroutines we should do `go tn.callback`. I dunno, there is probably a way to improve this.

Reply:
Oh, nevermind, you fixed it?

Reply:
No I didn't, I haven't really thought about how to do this in a way that works with the coroutines scheduler.